### PR TITLE
AC contribution change

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/6228

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Plan 1
- start with short contribution interval and staging
- claim grant
- add one verified publisher to monthly tip and set AC to 50 BAT
- restart browser
- you shouldn't receive notification about not enough funds

Plan 2
- start with short contribution interval and staging
- claim grant
- add multiple verified publishers to monthly tip so that you are over balance
- restart browser
- you should receive notification about not enough funds

Plan 3
- start with short contribution interval and staging
- claim grant
- add one verified publisher and multiple unverified publishers to that total is over balance
- restart browser
- you shouldn't receive notification about not enough funds

Plan 4
- start with short contribution interval and staging
- claim grant (30 BAT)
- set AC budget to 50 BAT
- add verified publisher to AC table
- add verified monthly tips publishers so that total is 20 BAT
- wait for contribution to happen
- monthly tips and AC should go through, where AC amount would be 10

Plan 4
- start with short contribution interval and staging
- claim grant (30 BAT)
- set AC budget to 20 BAT
- populate AC table with verified and unverified publishers
- wait for contribution to happen
- make sure that 20 BAT went through and that only verified publishers received BAT

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution.cc [hunk 1]
```diff
@@ -317,7 +222,7 @@ void Contribution::StartMonthlyContribution() {
                 _1,
                 _2));
 
-  SetTimer(&delay_ac_timer_id, 10);
+  SetTimer(&delay_ac_timer_id, 30);
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution.cc]: Change this to reduce potential race condition between recurring tips and AC. Will be fully fixed with https://github.com/brave/brave-browser/issues/6288

**Comment by ryanml** [comment]
Disregard, I see this failure on master

File: .storybook/locale.ts [hunk 2]
```diff
@@ -233,7 +232,7 @@ const locale: Record<string, string> = {
   siteBannerConnectedText: 'This creator has not yet signed up to receive contributions from Brave users. Your browser will keep trying to contribute until they verify, or until 90 days have passed.',
   sites: 'sites',
   tellOthers: 'Tell others about your tip.',
-  supportedSites: 'Supported Sites',
+  supportedSites: 'Sites viewed',
```
Comment (masparrow) [inline .storybook/locale.ts]: If this is a title, would expect 'Sites Viewed'

Comment (NejcZdovc) [reply .storybook/locale.ts]: it's not a title, it's a tab

**Comment by masparrow** [comment]
Code-wise - LGTM

Plan 1 & 3 seem ok for me - no notification.
Plan 2 - I didn't see any notification (tried twice - but could be me/user error)
Plan 4 (both of them :D) my monthly contribution date is set to Nov 7th... is there a way to advance this for testing?

**Comment by NejcZdovc** [comment]
@masparrow you need to set short interval for plans to work `npm start -- --rewards=reconcile-interval=3 --vmodule=*rewards*=6`

**Comment by LaurenWags** [comment]
Verified passed with

Brave | 0.72.102 Chromium: 78.0.3904.50 (Official Build) nightly (64-bit)
-- | --
Revision | 2accdc52c79976e264cd2694df6db31d1fccd8e8-refs/branch-heads/3904@{#658}
OS | macOS Version 10.13.6 (Build 17G5019)

- Verified test plan 1 worked as expected. Verified no message about not enough funds was displayed as BAT logo notification. Tip for verified, KYC'd publisher was contributed (confirmed tip on publishers dashboard).

- Verified test plan 2 worked as expected. BAT logo notification about not enough funds was displayed after browser restart.

- Verified test plan 3 worked as expected. Recurring monthly tips to unverified publishers were added to Pending Contributions list. Tip for verified, KYC'd publisher was contributed (confirmed tip on publishers dashboard). Verified no message about not enough funds was displayed as BAT logo notification.

- Verified test plan 4 worked as expected. Recurring monthly tips for 20 BAT were deducted from balance first. Remaining balance (10 BAT) was contributed to the single verified, KYC'd publisher in the list after. Confirmed tip and AC amounts on publishers dashboard.
TP4-ss


- Verified test plan 5 worked as expected. Auto Contribute went thru for 20 BAT. Only Verified publishers received BAT. Confirmed 20 BAT (less the 5% fee) AC amount on publisher dashboard. AC for unverified publishers were not listed in the Pending Contributions table.
TP5-ss

Also:
- Verified recurring monthly tips showed in Monthly Contributions panel on clean profile.
- Verified recurring monthly tips from previous version showed in Monthly Contributions panel on upgrade profile, not in Tips panel. Verified recurring monthly tips set up in previous profile were contributed after upgrade.
- Verified Text/Label changes as outlined in the spec. Logged #6483 for label changes that were missed/need to be updated. See slack for discussions.
label1
label2

**Pull request merged**